elm_kernel/kernel.py

```diff
@@ -173,6 +176,19 @@ def _send_success_result(self, javascript):
                 }
             })
 
+    def _copy_elm_package_file_to_tempdir(self):
+        """Copy elm-package.json to temporary directory where elm code is compiled
+        """
+        # existence of elm-package.json is not mandatory
+        if not os.path.isfile('elm-package.json'):
+            return
+
+        try:
```

Comment:
I think we should abort the compilation if we can't copy an extant `elm-package.json`. As written, it looks like we swallow the `PermissionException` and continue with the rest of the compilation.

I think what might be best is to not do any exception handling in this function. If any exceptions propagate to the top-level `do_compile()`, we notify the user with `_send_error_result()` and return an error code. I don't think detailed handling in functions like this add much to the diagnostics.